pkg/minikube/machine/fix.go

```diff
@@ -67,21 +67,24 @@ func fixHost(api libmachine.API, cc config.ClusterConfig, n config.Node) (*host.
 		return h, errors.Wrap(err, "Error loading existing host. Please try running [minikube delete], then run [minikube start] again.")
 	}
 
+	driverName := h.Driver.DriverName()
+
 	// check if need to re-run docker-env
-	maybeWarnAboutEvalEnv(cc.Driver, cc.Name)
+	maybeWarnAboutEvalEnv(driverName, cc.Name)
 
 	h, err = recreateIfNeeded(api, cc, n, h)
 	if err != nil {
 		return h, err
 	}
 
-	// Technically, we should only have to call provision if Docker has changed,
-	// but who can predict what shape the existing VM is in.
-	e := engineOptions(cc)
-	h.HostOptions.EngineOptions.Env = e.Env
-	err = provisionDockerMachine(h)
-	if err != nil {
-		return h, errors.Wrap(err, "provision")
+	if !driver.BareMetal(h.Driver.DriverName()) {
+		glog.Infof("%s is local, skipping re-provision as it requires SSH", driverName)
```

Comment:
this doesnt match the if logic

Reply:
Sorry, I attempted to turn a comment into a log message and made it only more confusing.